### ISSUE TITLE
DOC: warning against use of .values should show difference compared to to_numpy()/array 

### ISSUE THREAD

**Issue by davidgilbertson** [pandas-dev/pandas#48425]
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/version/1.4.0/reference/api/pandas.DataFrame.values.html

### Documentation problem

The docs state "We recommend using DataFrame.to_numpy() instead".

From the perspective of a beginner I see this and think "fine, I'll trust that this is for a good reason and use to_numpy() instead". But then I see in other documentation (e.g. PyTorch) the common use of `.values` and now I'm thinking "well is something going to break if I use `.values`? The Pandas people said don't do this! Should I change all usages of `.values` to `.to_numpy()`?"

So having a warning against something, but without a reason, is actually just a cause of confusion.

I see from the source that `to_numpy()` has some smarts about dtypes, which leads me do think that if I'm doing something like `torch.tensor(df.values, dtype=torch.float)` it's perfectly fine - I don't need Pandas to handle the dtypes because PyTorch will. Yet the big red warning in the docs leaves me with an uneasy feeling.

### Suggested fix for documentation

The addition of a comparison between `.values` and `.to_numpy()` with different outputs and an explanation as to what trouble a user could expect to see if they use `.values`.

In other words, what is the wisdom in the brain of the person who wrote the line "we recommend using to_numpy()"? Take that wisdom and put it on the page next to the warning.

**Comment by mroeschke** [comment]
Sounds reasonable to add examples showing the output of `.values` to `.to_numpy()/.array` when they differ.

**Comment by sqali** [comment]
Hi @mroeschke ,

I am new to open source and would love to work on this issue. Can you please guide me?

**Comment by sqali** [comment]
take
